src/main/java/org/breskul/connectivity/datasource/DataSourcePropertyResolver.java

```diff
@@ -0,0 +1,82 @@
+package org.breskul.connectivity.datasource;
+
+import lombok.extern.slf4j.Slf4j;
+import org.breskul.exception.BoboException;
+import org.breskul.exception.ExceptionMessage;
+
+import java.io.*;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static org.breskul.util.Constants.*;
+
+@Slf4j
+public class DataSourcePropertyResolver {
+    private Map<String, String> valueToPropertyName;
+
+    public DataSourcePropertyResolver() {
+        scanFile(DEFAULT_PROPERTIES_FILE_NAME);
+        initializeDataSource();
+    }
+
+    public DataSourcePropertyResolver(String fileName) {
+        scanFile(fileName);
+        initializeDataSource();
```

Comment:
I would not initialize datasource inside the property resolver. For me, it's different responsibility. Property resolver can be more common class (for all properties).

Reply:
- Extracted the connection pool and driver-agnostic logic to PooledDataSource
- Made PropertyResolver independent
- Added a bit documentation and logs